crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -439,13 +407,17 @@ async fn execute_transactions(
                 .await?;
         }
     }
+    authority_state
```

Comment:
To avoid code duplication, L402 ~ L420 can be calling execute_certificate_with_effects() then join_all() on all certificates and effects.

Reply:
Hm `execute_certificate_with_effects()` takes SignedTransactionEffects` or `CertifiedTransactionEffects`.. whereas the effects that StateSync persists are unsigned. Should we perhaps have `StateSync` persist the signed effects to at temp table for executor, as we do for the tx certs? cc @bmwill

Reply:
We shouldn't need signed effects as they're coming from a checkpoint

Reply:
Can we trust shared obj versions from unsigned effects though in the general case? cc @mwtian

Reply:
They are signed indirectly because of their inclusion in the checkpoint.